### PR TITLE
Added Theming Support

### PR DESCRIPTION
**Pull request by clintonb**
Views can now be themed by site. Credential renderings are now themed 
by program type, program, course, course run, and seat type.

LEARNER-516

### CODE REVIEW

**Comment by clintonb** [comment]
1. `credentials/base.html` now has three themable sections:
* Header
* Certificate
* Footer
2. Two new themes have been created for OpenEdX and edx.org.

**Comment by clintonb** [comment]
@edx/learner @edx/helio please review

**Comment by clintonb** [comment]
FYI @douglashall

File: credentials/apps/credentials/tests/test_views.py [hunk 1]
```diff
@@ -70,41 +73,35 @@ def _render_user_credential(self, use_proper_logo_url=True):
...
         response = self._render_user_credential()
+        self.assertEqual(response.status_code, 200)
 
-        self._assert_user_credential_template_data(response, self.user_credential, certificate_title=certificate_title)
-        self._assert_signatory_data(response, self.signatory_1)
-        self._assert_signatory_data(response, self.signatory_2)
+        expected = {
+            'user_credential': self.user_credential,
+            'user_data': self.MOCK_USER_DATA,
+            # TODO Determine how to test these values
```
Comment (rlucioni) [inline credentials/apps/credentials/tests/test_views.py]: Do you intend to come back to resolve this now?

File: credentials/apps/credentials/views.py [hunk 2]
```diff
@@ -34,67 +36,92 @@ def get_context_data(self, **kwargs):
...
+    @cached_property
+    def user_credential(self):
+        return get_object_or_404(
             UserCredential,
-            uuid=kwargs.get('uuid'),
+            uuid=self.kwargs.get('uuid'),
             status=UserCredential.AWARDED
         )
 
+    @cached_property
```
Comment (rlucioni) [inline credentials/apps/credentials/views.py]: I see you introducing a lot of these cached properties. The "cache" on these only persists for the lifetime of the instantiated class.  Are these methods being called repeatedly within the lifetime of the class? If not, these don't really add anything.

File: credentials/templates/credentials/base.html [hunk 3]
```diff
@@ -1,12 +1,52 @@
 {% extends 'base.html' %}
+{% load i18n %}
 {% load render_bundle from webpack_loader %}
 
 {% block title %}
+  {#  TODO: Determine title#}
```
Comment (rlucioni) [inline credentials/templates/credentials/base.html]: Is this resolved?

File: credentials/apps/core/migrations/0012_remove_siteconfiguration_verified_certificate_url.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('core', '0011_auto_20170410_1646'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
```
Comment (rlucioni) [inline credentials/apps/core/migrations/0012_remove_siteconfiguration_verified_certificate_url.py]: This field is currently in use in prod. Shouldn't you be deleting the field after your changes to stop using the field are deployed?

File: credentials/apps/credentials/tests/test_views.py (hunk 1 above)
Comment (clintonb) [inline credentials/apps/credentials/tests/test_views.py]: I would like to; but, I'm not holding out hope.

File: credentials/apps/credentials/views.py (hunk 2 above)
Comment (clintonb) [inline credentials/apps/credentials/views.py]: It was. Now it's not used. Removed.

File: credentials/templates/credentials/base.html (hunk 3 above)
Comment (clintonb) [inline credentials/templates/credentials/base.html]: Fixed.

File: credentials/apps/core/migrations/0012_remove_siteconfiguration_verified_certificate_url.py (hunk 4 above)
Comment (clintonb) [inline credentials/apps/core/migrations/0012_remove_siteconfiguration_verified_certificate_url.py]: Yes. Great catch!

File: credentials/apps/credentials/tests/test_views.py (hunk 1 above)
Comment (rlucioni) [inline credentials/apps/credentials/tests/test_views.py]: Okay. Maybe think twice about merging a commented out block to which we'll never return? I'd either fix it or just delete it and leave a TODO behind. Either way, not a big deal.

**Comment by mjfrey** [comment]
We should probably remove template from AbstractCertificate as it was never implemented.  This implementation will replace any notion of that.  

The model CertificateTemplate probably has no use.

**Comment by douglashall** [comment]
Thanks @clintonb! Can we get this up and running on a sandbox?

**Comment by clintonb** [comment]
@douglashall if you want to run the code use Docker devstack.

**Comment by douglashall** [comment]
@clintonb It would be great if people on the team who do not have the ability to set up devstack were able to see it in action.

**Comment by douglashall** [comment]
@clintonb Could you also provide an overview of how this theming functionality is used?

**Comment by clintonb** [comment]
@douglashall documentation is available at http://open-edx-credentials-guide.readthedocs.io/en/clintonb-theming-v2/theming.html. Everyone has the ability to run Docker devstack. If you really want see this, you're welcome to schedule some time on my calendar.

**Pull request merged**